tools/frr.in

```diff
@@ -202,36 +202,56 @@ stop()
 	if ! started "$inst" ; then
 		echo -n " ($inst)"
 		return 0
+
+		# Make sure that watch FRR is stopped first
+	elif [ $inst == "watchfrr" ]; then
+		PIDFILE=`pidfile $inst`
+		PID=`cat $PIDFILE 2>/dev/null`
+
+		# Send signal to die
+		kill -2 $PID 2>/dev/null &
+
+		# Ask waiter to... wait
+		stop_waiter $PID $inst
 	else
 		PIDFILE=`pidfile $inst`
 		PID=`cat $PIDFILE 2>/dev/null`
-		kill -2 $PID 2>/dev/null
-		#
-		#       Now we have to wait until $DAEMON has _really_ stopped.
-		#
+
+		# Send signal to die
+		kill -2 $PID 2>/dev/null &
+
+		# Ask waiter to... wait
+		stop_waiter $PID $inst &
+	fi
+}
+
+stop_waiter(){
+	PID=$1
+	inst=$2
+	if [ "$2" = "zebra" ]; then
+		echo -n "$inst. "
+		sleep 0.5
+		echo
+		echo "Stopping staticd since zebra is running. "
+		stop staticd
+	fi
+	if [ "$2" != "zebra" ]; then
 		if test -n "$PID" && kill -0 $PID 2>/dev/null; then
-			echo -n " (waiting) ."
 			cnt=0
 			while kill -0 $PID 2>/dev/null; do
 				cnt=`expr $cnt + 1`
-				if [ $cnt -gt 60 ]; then
+				if [ $cnt -gt 12000 ]; then
```

Comment:
This is 20 minutes, not 2 minutes as the comment indicates.  (It's sleeping for 0.1s)